compiler/rustc_type_ir/src/search_graph/stack.rs

```diff
@@ -26,6 +26,10 @@ pub(super) struct StackEntry<X: Cx> {
     /// The available depth of a given goal, immutable.
     pub available_depth: AvailableDepth,
 
+    /// Starts out as `None` and gets set when rerunning this
+    /// goal in case we encounter a cycle.
+    pub provisional_result: Option<X::Result>,
```

Comment:
moving this field as this type is now all the immutable fields and then the mutable ones